### ISSUE TITLE
Nanosecond not populated

### ISSUE THREAD

**Issue by coffeepots** [coffeepots/odbc#6]
Currently there is an issue where the `Fraction` part of `SQL_TIMESTAMP_STRUCT` is returning zero.
This is likely a precision issue when specifying the parameter's details.

**Comment by mprkins** [comment]
Is this fixed or still an issue ?

**Comment by coffeepots** [comment]
Unfortunately this is still an issue. Whilst the structure for ODBC is populated correctly, SQL Server doesn't return the nanoseconds portion of the datetime.

**Comment by gpdnet** [comment]
This doesn't seem to be an error in SQL Server or any ODBC driver. I have tested under Ubuntu 18.10 using both FreeTDS and the MS ODBC driver in plain C and the fractional time (milliceconds) is returned just fine. However, when using nim with the same ODBC (odbcsql) library calls and drivers the fraction for timestamp is not returned in the structure. This appears to be a nim->C struct mapping problem. The sizeof(SQL_TIMESTAMP_STRUCT) returns 24 when it should be 16! Perhaps some strange alignment or padding that nim is doing leading to the fraction member not being copied properly? To get round this in my code I return a SQL datetime type as a string and then parse into an nim DateTime type. BTW, with newer versions of nim the DateTime type can handle milliseconds. No need to use TimeInterval.

**Comment by coffeepots** [comment]
@gpdnet thanks very much for the insight, I've been puzzling over this on and off for a while. What you say makes a lot of sense. I don't want to pull the datetime over as a string if I can possibly avoid it, but might be worth it for the interim so that the functionality is present.

Newer versions of Nim do support millisecond times and there's been a few quality of life improvements lately. I do plan to replace `TimeInterval` with `Time`.

**Comment by gpdnet** [comment]
I've filed a bug report with the nim project itself for odbcsql. It looks like the type (size) of TSqlInteger and SqlUInteger is wrong. It should be fixed at 32 bits.

In the mean time I've defined a slightly modified struct locally in my program:
type
   SQL_TIMESTAMP_STRUCT1 {.final, pure.} = object
      Year*: SqlUSmallInt
      Month*: SqlUSmallInt
      Day*: SqlUSmallInt
      Hour*: SqlUSmallInt
      Minute*: SqlUSmallInt
      Second*: SqlUSmallInt
      Fraction*: int32

Using this struct in the SqlBindCol to fetch a timestamp works like a charm. A reasonable workaround until the standard library for odbcsql is fixed (hopefully).
